modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -0,0 +1,133 @@
+	struct lvgl_disp_data *disp_data = disp->driver->user_data;
+	struct display_capabilities *cap = &disp_data->cap;
+
+	switch (evt->code) {
+	case INPUT_ABS_X:
+		data->pending_event.point.x = evt->value;
+		break;
+	case INPUT_ABS_Y:
+		data->pending_event.point.y = evt->value;
+		break;
+	case INPUT_BTN_TOUCH:
+		data->pending_event.state = evt->value ? LV_INDEV_STATE_PR : LV_INDEV_STATE_REL;
+		break;
+	}
+
+	if (!evt->sync) {
+		return;
+	}
+
+	/* adjust coordinates */
+	if (cfg->swap_xy) {
+		lv_coord_t x;
+
+		x = data->pending_event.point.x;
+		data->pending_event.point.x = data->pending_event.point.y;
+		data->pending_event.point.y = x;
+	}
+
+	if (cfg->invert_x) {
+		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
+		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+			data->pending_event.point.x =
+				cap->x_resolution - data->pending_event.point.x;
+		} else {
+			data->pending_event.point.x =
+				cap->y_resolution - data->pending_event.point.x;
+		}
+	}
+
+	if (cfg->invert_y) {
+		if (cap->current_orientation == DISPLAY_ORIENTATION_NORMAL ||
+		    cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+			data->pending_event.point.y =
+				cap->y_resolution - data->pending_event.point.y;
+		} else {
+			data->pending_event.point.y =
+				cap->x_resolution - data->pending_event.point.y;
+		}
+	}
+
+	/* rotate touch point to match display rotation */
+	if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_90) {
+		lv_coord_t x;
+
+		x = data->pending_event.point.x;
+		data->pending_event.point.x = data->pending_event.point.y;
+		data->pending_event.point.y = cap->y_resolution - x;
+	} else if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_180) {
+		data->pending_event.point.x = cap->x_resolution - data->pending_event.point.x;
+		data->pending_event.point.y = cap->y_resolution - data->pending_event.point.y;
+	} else if (cap->current_orientation == DISPLAY_ORIENTATION_ROTATED_270) {
+		lv_coord_t x;
+
+		x = data->pending_event.point.x;
+		data->pending_event.point.x = cap->x_resolution - data->pending_event.point.y;
+		data->pending_event.point.y = x;
+	}
+
+	/* filter readings within display */
+	if (data->pending_event.point.x <= 0) {
+		data->pending_event.point.x = 0;
+	} else if (data->pending_event.point.x >= cap->x_resolution) {
+		data->pending_event.point.x = cap->x_resolution - 1;
+	}
+
+	if (data->pending_event.point.y <= 0) {
+		data->pending_event.point.y = 0;
+	} else if (data->pending_event.point.y >= cap->y_resolution) {
+		data->pending_event.point.y = cap->y_resolution - 1;
+	}
+
+	if (k_msgq_put(cfg->common_config.event_msgq, &data->pending_event, K_NO_WAIT) != 0) {
+		LOG_DBG("Could not put input data into queue");
+	}
+}
+
+int lvgl_pointer_input_init(const struct device *dev)
+{
+	return register_lvgl_indev_driver(LV_INDEV_TYPE_POINTER, dev);
+}
+
+#define LVGL_POINTER_INPUT_DEFINE(_inst)                                                           \
```

Comment:
Could you just use `_inst`? don't think there's any naming conflict in here

Reply:
Is this a typo? Did you mean changing the macro parameter name from `_inst` to `inst`?

Reply:
yeah sorry I mean use `inst`